src/app/client/src/app/modules/notes/components/note-list/note-list.component.ts

```diff
@@ -0,0 +1,247 @@
+   */
+  selectedNote: INotesListData;
+  /**
+   * Stores the index of the selected note in notesList array.
+   */
+  selectedIndex = 0;
+  /**
+   * This variable stores the search input from the search bar.
+   */
+  searchData: string;
+  /**
+   * The course id of the selected course.
+   */
+  courseId: string;
+  /**
+   * The content id of the selected course.
+   */
+  contentId: string;
+  /**
+   * This variable helps in displaying and dismissing the delete modal.
+   */
+  showDelete: false;
+  /**
+   * user id from user service.
+   */
+  userId: string;
+  /**
+   * To get course and note params.
+   */
+  private activatedRoute: ActivatedRoute;
+  /**
+   * To display toaster(if any) after each API call.
+   */
+  private toasterService: ToasterService;
+  /**
+   * Reference of notes service.
+   */
+  noteService: NotesService;
+  /**
+   * Reference of user service.
+   */
+  userService: UserService;
+  /**
+   * Reference of content service.
+   */
+  contentService: ContentService;
+  /**
+   * Reference of resource service.
+   */
+  resourceService: ResourceService;
+  /**
+   * Reference of modal service.
+   */
+  modalService: SuiModalService;
+  route: Router;
+  routerNavigationService: RouterNavigationService;
+  /**
+   * The constructor
+   *
+   * @param {ToasterService} toasterService Reference of toasterService.
+   * @param {UserService} userService Reference of userService.
+   * @param {ContentService} contentService Reference of contentService.
+   * @param {ResourceService} resourceService Reference of resourceService.
+   * @param {SuiModalService} modalService Reference of modalService.
+   * @param {NotesService} noteService Reference of notesService.
+   * @param {ActivatedRoute} activatedRoute Reference of activatedRoute.
+   */
+
+  constructor(noteService: NotesService,
+    userService: UserService,
+    contentService: ContentService,
+    resourceService: ResourceService,
+    modalService: SuiModalService,
+    activatedRoute: ActivatedRoute,
+    toasterService: ToasterService,
+    route: Router,
+    routerNavigationService: RouterNavigationService) {
+    this.toasterService = toasterService;
+    this.activatedRoute = activatedRoute;
+    this.userService = userService;
+    this.noteService = noteService;
+    this.contentService = contentService;
+    this.resourceService = resourceService;
+    this.modalService = modalService;
+    this.route = route;
+    this.routerNavigationService = routerNavigationService;
+
+  }
+  /**
+   * To initialize notesList and showDelete.
+   * Listens to updateNotesListData event as well.
+   */
+  ngOnInit() {
+
+    this.showDelete = false;
+
+    /**
+     * Initializing selectedNote
+    */
+    this.noteService.updateNotesListData.subscribe(data => {
+      if (data === 0) {
+        this.selectedIndex = 0;
+        this.notesList = this.notesList.filter((note) => {
+          return note.id !== this.selectedNote.id;
```

Comment:
On first run of this code, selectedNote is not even initialised. Why is this filter code required?